stacker/src/memory_arena.rs

```diff
@@ -376,4 +378,20 @@ mod tests {
         assert_eq!(arena.read::<MyTest>(addr_a), a);
         assert_eq!(arena.read::<MyTest>(addr_b), b);
     }
+
+    #[test]
+    fn test_can_allocate_4GB() {
```

Comment:
Would it be worth a test that goes a byte past the limit and `should_panic` (or fails gracefully, if that's the API), so that we are sure that we know what it is? Or is just restoring the previous limit fine?

Reply:
done.